ansible/roles/kubernetes/tasks/secrets.yml

```diff
@@ -28,6 +28,19 @@
 - include: gen_certs.yml
   when: inventory_hostname == groups['masters'][0]
 
+- name: Archive certificates
```

Comment:
Option 2 of what you wrote. Currently, the certificates (server.crt/key and kubecfg.crt/key) are generated on the first master. This needs to be shared across all the masters, so I tar it, fetch it to the ansible host, then unpack it on each master.